app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/EditEvent.kt

```diff
@@ -0,0 +1,311 @@
+
+    val selectedLocation: MutableState<Location?> = remember { mutableStateOf(event!!.location) }
+    val tags = remember { mutableStateOf(event!!.tags) }
+    val showPopup = remember { mutableStateOf(false) }
+
+    var imageUri by remember { mutableStateOf<Uri?>(null) }
+    var imageBitmap by remember { mutableStateOf<ImageBitmap?>(null) }
+
+    val imagePickerLauncher =
+        rememberLauncherForActivityResult(contract = ActivityResultContracts.GetContent()) {
+            uri: Uri? ->
+          imageUri = uri
+          uri?.let { uriNonNull ->
+            val inputStream: InputStream? =
+                try {
+                  context.contentResolver.openInputStream(uriNonNull)
+                } catch (e: Exception) {
+                  e.printStackTrace()
+                  null
+                }
+            inputStream?.let {
+              val bitmap = BitmapFactory.decodeStream(it)
+              imageBitmap = bitmap.asImageBitmap()
+            }
+          }
+        }
+
+    val textFieldColors =
+        TextFieldDefaults.colors(
+            focusedTextColor = MaterialTheme.colorScheme.onBackground,
+            unfocusedTextColor = MaterialTheme.colorScheme.onBackground,
+            unfocusedContainerColor = Color.Transparent,
+            focusedContainerColor = Color.Transparent,
+            cursorColor = MaterialTheme.colorScheme.outlineVariant,
+            focusedLabelColor = MaterialTheme.colorScheme.tertiary,
+            unfocusedLabelColor = MaterialTheme.colorScheme.tertiary,
+            focusedIndicatorColor = MaterialTheme.colorScheme.tertiary,
+            unfocusedIndicatorColor = MaterialTheme.colorScheme.tertiary)
+
+    Scaffold(
+        topBar = {
+          Row(verticalAlignment = Alignment.CenterVertically) {
+            IconButton(onClick = { nav.goBack() }) {
+              Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Go back")
+            }
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                text = "Done",
+                style = MaterialTheme.typography.bodyLarge.copy(fontWeight = FontWeight.SemiBold),
+                modifier =
+                    Modifier.padding(end = 15.dp).clickable {
+                      if (titleState.value.isNotEmpty() &&
+                          descriptionState.value.isNotEmpty() &&
+                          locationState.value.isNotEmpty() &&
+                          priceText.isNotEmpty() &&
+                          url.value.isNotEmpty()) {
+                        val updatedEvent =
+                            event!!.copy(
+                                title = titleState.value,
+                                description = descriptionState.value,
+                                location = selectedLocation.value!!,
+                                date = pickedDate.value,
+                                time = pickedTime.value,
```

Comment:
When we click on Done, the time on EventInfo looks like this:

<img width="169" alt="Capture d’écran 2024-05-21 à 17 34 31" src="https://github.com/SwEnt-Project-G18/SwEntApp/assets/92448776/c6bf6abb-487a-4847-94a0-015ea510fc08">

Reply:
Maybe these methods in the data class event could be helpful :
`/**
 * Converts the time of an event to a string.
 *
 * @param eventTime The event's time.
 * @return The string representation of the time.
 */
fun getEventTimeString(eventTime: LocalTime): String {
  val minutes = if (eventTime.minute < 9) "0${eventTime.minute}" else eventTime.minute
  val hours = if (eventTime.hour < 9) "0${eventTime.hour}" else eventTime.hour
  return "${hours}:${minutes}"
}

/**
 * Converts the date of an event to a string.
 *
 * @param eventDate The event's date.
 * @return The string representation of the date.
 */
fun getEventDateString(eventDate: LocalDate): String {
  val date =
      if (eventDate == LocalDate.now()) "Today"
      else "${eventDate.dayOfMonth}/${eventDate.monthValue}/${eventDate.year}"
  return date
}`

Reply:
True, the error came from InitFunctions, good catch